verification/src/transaction_verifier.rs

```diff
@@ -183,7 +183,9 @@ impl<'a> EmptyVerifier<'a> {
     }
 
     pub fn verify(&self) -> Result<(), Error> {
-        if self.transaction.is_empty() {
+        if self.transaction.inputs().is_empty()
+            || (self.transaction.outputs().is_empty() && !self.transaction.is_cellbase())
+        {
```

Comment:
After this change, `TransactionView::is_empty` is useless. We can remove this function now.